src/terrain/material.hpp

```diff
@@ -98,4 +99,19 @@ class TerrainColorMapping {
     };
 };
 
+[[nodiscard]] inline bool
+material_in(
+    const std::set<std::pair<MaterialId, ColorId>> materials, MaterialId material_id,
+    ColorId color_id
+) {
+    for (auto element : materials) {
+        if ((element.first == MAT_ANY_MATERIAL || element.first == material_id)
+            && (element.second == COLOR_ANY_COLOR || element.second == color_id)) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
```

Comment:
```suggestion
[[nodiscard]] inline bool
material_in(
    const std::set<std::pair<MaterialId, ColorId>> materials, MaterialId material_id,
    ColorId color_id
) {
    auto same_mat = [&material_id](MaterialId m) {
        return m == MAT_ANY_MATERIAL || m == material_id;
    };
    
    auto same_color = [&color_id](ColorId c) {
        return c == COLOR_ANY_COLOR || c == color_id;
    };
    
    for (const auto& [mat, color] : materials) {
        if (same_mat(mat) && same_color(color))
            return true;
    }

    return false;
}

```

yes this gets compiled down to the same code on `-O3`: https://godbolt.org/z/nhjYon8cz

Also why `std::set` instead of `std::unordered_set`.